### PR TITLE
Add owners file

### PR DESCRIPTION
**Pull request by jotak**

### CODE REVIEW

**Comment by msherif1234** [comment]
Hi @jotak, I don't think u need the entire team to be in the approval/reviewer list for the agent repo, maybe u and stlee for now ?

**Comment by jotak** [comment]
At some point we should add a `GOVERNANCE.md` file to each project to clarify maintainership, emeritus contributors, etc. For the time being I'm just reproducing in prow what we have in our github maintainers team.

**Comment by jotak** [comment]
Example of a governance file: https://github.com/jaegertracing/jaeger/blob/main/GOVERNANCE.md

**Review by OlivierCazade — APPROVED** [review]

**Pull request merged**
